implementations/rust/ockam/ockam_command/src/identity/default.rs

```diff
@@ -27,23 +28,34 @@ impl DefaultCommand {
 }
 
 fn run_impl(opts: CommandGlobalOpts, cmd: DefaultCommand) -> miette::Result<()> {
-    let state = opts.state.identities;
-    let idt = state.get(&cmd.name)?;
-    // If it exists, warn the user and exit
-    if state.is_default(idt.name())? {
-        Err(miette!(
-            "The identity '{}' is already the default",
-            &cmd.name
-        ))
-    }
-    // Otherwise, set it as default
-    else {
-        state.set_default(idt.name())?;
-        opts.terminal
-            .stdout()
-            .plain(fmt_ok!("The identity '{}' is now the default", &cmd.name))
-            .machine(&cmd.name)
-            .write_line()?;
+    if let Some(name) = cmd.name {
+        let state = opts.state.identities;
+        let idt = state.get(&name)?;
+        // If it exists, warn the user and exit
+        if state.is_default(idt.name())? {
+            Err(miette!(
+                "The identity '{}' is already the default",
+                &name
+            ))
+        }
+        // Otherwise, set it as default
+        else {
+            state.set_default(idt.name())?;
+            opts.terminal
+                .stdout()
+                .plain(fmt_ok!("The identity '{}' is now the default", &name))
+                .machine(&name)
+                .write_line()?;
+            Ok(())
+        }
+    }else {
+        // no argument provided
+        initialize_identity_if_default(&opts, &None);
+        let name = get_identity_name(&opts.state, &None);
+        let state = opts.state.identities.get(&name)?;
+        let identifier = state.config().identifier();
+        let identifier_display = IdentifierDisplay(identifier);
+        opts.terminal.stdout().plain(&format!("{}", identifier_display)).write_line()?;
```

Comment:
Since the CLI user uses identities by passing their associated names, this should be:

```
opts.terminal.stdout().plain(&fmt_ok!("The default identity name is {}", state.name())).write_line()?;
```

Reply:
Updated this